### PR TITLE
Separate grpc transcoding library

### PR DESCRIPTION
**Pull request by rkpagadala**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# grpc-transcoding
+
+grpc-transcoding is a library that supports [transcoding](https://cloud.google.com/endpoints/docs/transcoding) so that HTTP/JSON can be converted to gRPC.
+
+It helps you to provide your APIs in both gRPC and RESTful style at the same time. The code is used in istio [proxy](https://github.com/istio/proxy) to provide HTTP+JSON interface to gRPC service.
+
+The code can be built with the following command
+
+bazel build //...
```
Comment (lizan) [inline README.md]: wrap with ```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+Tests can be run using
+
+bazel test //...
```
Comment (lizan) [inline README.md]: ditto

File: README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# grpc-transcoding
+
+grpc-transcoding is a library that supports [transcoding](https://cloud.google.com/endpoints/docs/transcoding) so that HTTP/JSON can be converted to gRPC.
+
+It helps you to provide your APIs in both gRPC and RESTful style at the same time. The code is used in istio [proxy](https://github.com/istio/proxy) to provide HTTP+JSON interface to gRPC service.
```
Comment (lizan) [inline README.md]: link to Cloud Endpoints as well

File: README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
```
Comment (lizan) [inline README.md]: instruction for installing bazel (a link is enough)

File: WORKSPACE [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    "googleapis_repositories",
+)
+
+protobuf_repositories()
+
+googletest_repositories()
+
+googleapis_repositories()
+
+# Workaround for Bazel > 0.4.0 since it needs newer protobuf.bzl from:
```
Comment (lizan) [inline WORKSPACE]: this is no longer used, remove line 31-41

File: WORKSPACE [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+#
+
+load(
+    "//src:repositories.bzl",
```
Comment (lizan) [inline WORKSPACE]: repositories.bzl can live outside of src dir.

Comment (rkpagadala) [inline WORKSPACE]: Done

File: WORKSPACE (hunk 5 above)
Comment (rkpagadala) [inline WORKSPACE]: Done

File: README.md (hunk 1 above)
Comment (rkpagadala) [inline README.md]: Done

File: README.md (hunk 2 above)
Comment (rkpagadala) [inline README.md]: Done

File: README.md (hunk 4 above)
Comment (rkpagadala) [inline README.md]: Done

File: README.md (hunk 3 above)
Comment (rkpagadala) [inline README.md]: Done

**Comment by rkpagadala** [comment]
PTAL

**Pull request merged**